tinyagentos/projects/task_store.py

```
@@ -373,8 +383,14 @@ async def update_task(
             sets.append("claimed_at = ?"); params.append(None); patch["claimed_at"] = None
         sets.append("updated_at = ?"); params.append(time.time())
         params.append(task_id)
+        where = "id = ?"
+        if status is not None:
+            # Close the read-then-write gap on the parked guard above: if the
+            # task was parked between that read and this write, the status edit
+            # must not land (it was decided against a stale row).
+            where += " AND status != 'parked'"
         await self._db.execute(
-            f"UPDATE project_tasks SET {', '.join(sets)} WHERE id = ?", params
+            f"UPDATE project_tasks SET {', '.join(sets)} WHERE {where}", params
```

Ruff (0.16.3)

[error] Possible SQL injection vector through string-based query construction

(S608)